app/pages/register/beneficiaries/other/mld5/UKResidentYesNoPage.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pages.register.beneficiaries.other.mld5
+
+import models.UserAnswers
+import play.api.libs.json.JsPath
+import pages.QuestionPage
+import pages.register.beneficiaries.other.{AddressInternationalPage, AddressUKPage}
+import sections.beneficiaries.{Beneficiaries, OtherBeneficiaries}
+
+import scala.util.Try
+
+final case class UKResidentYesNoPage(index: Int) extends QuestionPage[Boolean] {
+
+  override def path: JsPath = JsPath \ Beneficiaries \ OtherBeneficiaries \ index \ toString
+
+  override def toString: String = "ukResidentYesNo"
+
+  override def cleanup(value: Option[Boolean], userAnswers: UserAnswers): Try[UserAnswers] = {
+    value match {
+      case Some(true) => userAnswers.remove(AddressInternationalPage(index))
+      case Some(false) => userAnswers.remove(AddressUKPage(index))
```

Comment:
Is this correct? I would have thought answering `yes` would cleanup country of residency and `no` would require no cleanup